src/surface_conditions/surface_conditions.jl

```diff
@@ -21,25 +21,66 @@ function update_surface_conditions!(Y, p, t)
     )
     int_local_geometry_values =
         Fields.field_values(Fields.level(Fields.local_geometry_field(Y.c), 1))
-
     (; ᶜts, ᶜu, sfc_conditions, sfc_setup, params, atmos) = p
     int_ts_values = Fields.field_values(Fields.level(ᶜts, 1))
     int_u_values = Fields.field_values(Fields.level(ᶜu, 1))
     int_z_values =
         Fields.field_values(Fields.level(Fields.coordinate_field(Y.c).z, 1))
     sfc_conditions_values = Fields.field_values(sfc_conditions)
 
+    is_prognostic_sfc = atmos.surface_model isa PrognosticSurfaceTemperature
+    sfc_prognostic_temp =
+        is_prognostic_sfc ? Fields.field_values(Y.sfc.T) : nothing
+
     if sfc_setup isa Function
-        @. sfc_conditions_values = surface_state_to_conditions(
-            sfc_setup(sfc_local_geometry_values.coordinates, int_z_values, t),
-            sfc_local_geometry_values,
-            int_ts_values,
-            projected_vector_data(CT1, int_u_values, int_local_geometry_values),
-            projected_vector_data(CT2, int_u_values, int_local_geometry_values),
-            int_z_values,
-            params,
-            atmos,
-        )
+        if (is_prognostic_sfc)
+            @. sfc_conditions_values = surface_state_to_conditions(
+                sfc_setup(
+                    sfc_local_geometry_values.coordinates,
+                    int_z_values,
+                    t,
+                ),
+                sfc_local_geometry_values,
+                int_ts_values,
+                projected_vector_data(
+                    CT1,
+                    int_u_values,
+                    int_local_geometry_values,
+                ),
+                projected_vector_data(
+                    CT2,
+                    int_u_values,
+                    int_local_geometry_values,
+                ),
+                int_z_values,
+                params,
+                atmos,
+                sfc_prognostic_temp,
```

Comment:
This function call appears to have too many arguments

Reply:
I added a default argument to the surface_state_to_conditions() function, which is why the function calls have a different number of arguments. Is this okay with you?

Reply:
Adding the argument is fine, can we reduce the code duplication in this if-else block?

(
```julia
                sfc_setup(
                    sfc_local_geometry_values.coordinates,
                    int_z_values,
                    t,
                ),
                sfc_local_geometry_values,
                int_ts_values,
                projected_vector_data(
                    CT1,
                    int_u_values,
                    int_local_geometry_values,
                ),
                projected_vector_data(
                    CT2,
                    int_u_values,
                    int_local_geometry_values,
                ),
                int_z_values,
                params,
                atmos,
                sfc_prognostic_temp,
```
)

Reply:
sounds good, thanks Charles!

Reply:
I think we had this before, but it resulted in allocations?

Reply:
I think it did, but I can try to reduce code duplication in other ways. Could we talk about it tomorrow?

Reply:
Sorry, I'm not sure I'll have time today. Shouldn't we be able to just use dispatch here?

Reply:
@charleskawczynski Since this PR is so long, do you think I could merge it, then open another PR with multiple dispatch?